enterprise/internal/codeintel/resolvers/position.go

```diff
@@ -0,0 +1,160 @@
+package resolvers
+
+import (
+	"context"
+	"io"
+	"strings"
+
+	"github.com/sourcegraph/go-diff/diff"
+	"github.com/sourcegraph/go-lsp"
+	"github.com/sourcegraph/sourcegraph/cmd/frontend/backend"
+	"github.com/sourcegraph/sourcegraph/cmd/frontend/types"
+	"github.com/sourcegraph/sourcegraph/internal/vcs/git"
+)
+
+type positionAdjuster struct {
+	hunks []*diff.Hunk
+}
+
+// newPositionAdjuster creates a positionAdjuster by performing a git diff operation on the give
+// path between the source and target commits. If the commits are the same then a trivial-case
+// no-op position adjuster is returned.
+func newPositionAdjuster(
+	ctx context.Context,
+	repo *types.Repo,
+	sourceCommit string,
+	targetCommit string,
+	path string,
+) (*positionAdjuster, error) {
+	if sourceCommit == targetCommit {
+		return &positionAdjuster{hunks: nil}, nil
+	}
+
+	cachedRepo, err := backend.CachedGitRepo(ctx, repo)
+	if err != nil {
+		return nil, err
+	}
+
+	reader, err := git.ExecReader(
+		ctx,
+		*cachedRepo,
+		[]string{"diff", sourceCommit, targetCommit, "--", path},
+	)
```

Comment:
Optional formatting suggestion:

```suggestion
	reader, err := git.ExecReader(ctx, *cachedRepo, []string{
		"diff", sourceCommit, targetCommit, "--", path,
	})
```